File: ScheduleWebApp/ClientApp/src/app/_models/lesson.ts
import { Discipline } from './discipline';
import { DisciplineType } from './discipline-type';
import { Teacher } from './teacher';
import { Cabinet } from './cabinet';
import { Group } from './group';

export class Lesson {
  isExists: boolean;
  discipline: Discipline;
  disciplineType: DisciplineType;
  cabinet: Cabinet;
  teacher: Teacher;
  group: Group;

  constructor() {
    this.isExists = false;
    this.discipline = new Discipline(null, null);
    this.disciplineType = new DisciplineType(null, null);
    this.cabinet = new Cabinet(null, null);
    this.teacher = new Teacher(null, null, null, null, null);
    this.group = new Group(null, null);
  }
}

File: Database/Entities/Lesson.cs
using System.ComponentModel.DataAnnotations;

namespace Database.Entities
{
    public class Lesson
    {
        public bool isExists { get; set; }
        public Discipline Discipline { get; set; }
        public DisciplineType DisciplineType { get; set; }
        public Cabinet Cabinet { get; set; }
        public Teacher Teacher { get; set; }
        public StudyGroup Group { get; set; }

        public Lesson() {
            isExists = false;
        }

        public Lesson(
            Discipline discipline,
            DisciplineType disciplineType,
            Cabinet cabinet, 
            Teacher teacher,
            StudyGroup studyGroup
        ) {
            isExists = true;
            Discipline = discipline;
            DisciplineType = disciplineType;
            Cabinet = cabinet;
            Teacher = teacher;
            Group = studyGroup;
        }

    }
}
